src/main/java/org/carlspring/cloud/storage/s3fs/S3Path.java

```diff
@@ -704,7 +730,7 @@ private String encode(String uri)
     {
         // remove special case URI starting with //
         uri = uri.replace("//", "/");
```

Comment:
The fork that is inside the Cromwell code base does a full URI encoding. It can cause issues with other software that assume that a s3 path is a URI (classic example that cause me to need to change this `s3://bucket/folder/folder#2/object` is a URI that needs to be encoded but it is a valid path.

Reply:
@elerch , @markjschreiber,

Would either of you fine chaps like to raise a follow-up issue for this in our issue tracker with proposals on how we should address this? Thanks! :)